src/include/duckdb/function/udf_function.hpp

```diff
@@ -0,0 +1,317 @@
+				break;
+			case 3:
+				CreateTernaryFunction<TR, Args...>(name, args, ret_type, udf_func);
+				break;
+			default:
+				throw duckdb::Exception(ExceptionType::EXECUTOR, "UDF function only supported until ternary!");
+		}
+	}
+
+private:
+	ClientContext &_context;
+
+	//-------------------------------- Templated functions --------------------------------//
+	template<typename TR, typename... Args>
+	void CreateUnaryFunction(string name, TR (*udf_func)(Args...)) {
+		assert(sizeof...(Args) == 1);
+		CreateUnaryFunction<TR, Args...>(name, udf_func);
+	}
+
+	template<typename TR, typename TA>
+	void CreateUnaryFunction(string name, TR (*udf_func)(TA)) {
+	    udf_function_t udf_function = [=] (DataChunk &input, ExpressionState &state, Vector &result) -> void {
+										UnaryExecutor::Execute<TA, TR>(input.data[0],
+																	   result,
+																	   input.size(),
+																	   udf_func);
+									};
+	    RegisterFunction<TR, TA>(name, udf_function);
+	}
+
+	template<typename TR, typename... Args>
+	void CreateBinaryFunction(string name, TR (*udf_func)(Args...)) {
+		assert(sizeof...(Args) == 2);
+		CreateBinaryFunction<TR, Args...>(name, udf_func);
+	}
+
+	template<typename TR, typename TA, typename TB>
+	void CreateBinaryFunction(string name, TR (*udf_func)(TA, TB)) {
+	    udf_function_t udf_function = [=] (DataChunk &input, ExpressionState &state, Vector &result) -> void {
+										BinaryExecutor::Execute<TA, TB, TR>(input.data[0],
+																		   	input.data[1],
+																		    result,
+																		    input.size(),
+																		    udf_func);
+									};
+	    RegisterFunction<TR, TA, TB>(name, udf_function);
+	}
+
+	template<typename TR, typename... Args>
+	void CreateTernaryFunction(string name, TR (*udf_func)(Args...)) {
+		assert(sizeof...(Args) == 3);
+		CreateTernaryFunction<TR, Args...>(name, udf_func);
+	}
+
+	template<typename TR, typename TA, typename TB, typename TC>
+	void CreateTernaryFunction(string name, TR (*udf_func)(TA, TB, TC)) {
+	    udf_function_t udf_function = [=] (DataChunk &input, ExpressionState &state, Vector &result) -> void {
+										TernaryExecutor::Execute<TA, TB, TC, TR>(input.data[0],
+																		   	   	 input.data[1],
+																		   	   	 input.data[2],
+																				 result,
+																				 input.size(),
+																				 udf_func);
+									};
+	    RegisterFunction<TR, TA, TB, TC>(name, udf_function);
+	}
+
+	template<typename TR, typename... Args>
+	void RegisterFunction(string name, udf_function_t udf_function) {
+	    vector<SQLType> arguments;
+	    GetArgumentTypesRecursive<Args...>(arguments);
+
+	    SQLType ret_type = GetArgumentType<TR>();
+
+		ScalarFunction scalar_function = ScalarFunction(name, arguments, ret_type, nullptr, false,
+														nullptr, nullptr, udf_function);
+		CreateScalarFunctionInfo info(scalar_function);
+
+		_context.transaction.BeginTransaction();
```

Comment:
This should not be done here. Transaction management should rather be left up to the user. This will also fail if the user is already inside a transaction. Rather, check out `ClientContext::RunFunctionInTransaction` that can be used to correctly start/commit a transaction in auto-commit mode.